app/models/collection.rb

```diff
@@ -124,46 +126,76 @@ def add_granule(current_user)
     granules_count = 1
     native_format = self.records[0].format
 
-      #checking that no granule exists for previously imported/deleted records.
-      if self.granules.count == 0
-        #only selecting granules for certain formats per business rules
-        if Collection::INCLUDE_GRANULE_FORMATS.include? native_format
-            #creating all the Granule related objects
-            granules_components = Granule.assemble_granule_components(self.concept_id, granules_count, self, current_user)
+
+    #only selecting granules for certain formats per business rules
+    if Collection::INCLUDE_GRANULE_FORMATS.include? native_format
+      #creating all the Granule related objects
+      granules_components = Granule.assemble_granule_components(concept_id, granules_count, self, current_user)
+    end
+
+    #saving all the related collection and granule data in a combined transaction
+    granules_components.flatten.each {|savable_object|
```

Comment:
So I looked into this why save! is being called and it seems like at one time they tried to do a transaction to handle all the saves being done under one transaction, but then for some reason pulled this out.   You can check git history to see that.   I'd like to check and see if there is any comments why they took it out.